### PR TITLE
Feat: Added disable merch toggle button

### PR DESCRIPTION
**Pull request by limivann**
## Overview

This PR adds a disable merch store button at the admin panel. The button will disable the merch store in ntuscse website. Note that the backend has not been implemented. 

Updates:
Added route checking for merch store status on merch routes

### CODE REVIEW

File: apps/web/pages/merch/index.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        setLoading(false);
+      } catch (error) {
+        // TODO: display error
+        setLoading(false);
+      }
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    fetchMerchSaleStatus();
+  }, []);
```
Comment (limivann) [inline apps/web/pages/merch/index.tsx:32]: @chanbakjsd Please evaluate if we should use useQuery for this fetch

File: apps/cms/src/admin/views/MerchOverview.tsx [hunk 2]
```diff
@@ -19,6 +59,29 @@ const MerchOverview: AdminView = ({ user, canAccessAdmin }) => {
...
+        <textarea
+          value={displayText}
+          onChange={(e: ChangeEvent<HTMLTextAreaElement>) =>
+            setDisplayText(e.target.value)
+          }
+          placeholder="Enter display text"
+          rows={4}
+          cols={50}
+        />
+      )}
```
Comment (limivann) [inline apps/cms/src/admin/views/MerchOverview.tsx:75]: @Iyzyman CharkraUI is not installed for payload, how do we proceed with the styling

File: apps/cms/src/admin/utils/RenderCellFactory.tsx [hunk 3]
```diff
@@ -2,42 +2,122 @@ import React from "react";
...
+          children,
+        }) => (
+          <span>
+            {children.map((imageUrl: string, index: number) => (
+              <img
+                key={index}
+                src={imageUrl}
+                alt={`image ${index + 1}`}
+                style={{ paddingTop: 10, paddingBottom: 10 }}
+              />
```
Comment (limivann) [inline apps/cms/src/admin/utils/RenderCellFactory.tsx:34]: @Iyzyman Linting issue after pull from main.

**Review by Iyzyman — DISMISSED** [review]

File: apps/web/pages/merch/index.tsx (hunk 1 above)
Comment (chanbakjsd) [reply apps/web/pages/merch/index.tsx:32]: Yes, please use `useQuery` for this if possible as it is an API call.

**Review by Iyzyman — APPROVED** [review]

**Review by chanbakjsd — APPROVED** [review]
LGTM 🚀

**Review by Iyzyman — APPROVED** [review]

**Review by jack-thant — APPROVED** [review]
LGTM

**Review by Iyzyman — DISMISSED** [review]

**Pull request merged**